File: core/src/com/colorflow/utility/effect/ExplosionPool.java
package com.colorflow.utility.effect;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.graphics.g2d.ParticleEffect;
import com.badlogic.gdx.scenes.scene2d.Actor;
import com.badlogic.gdx.scenes.scene2d.Stage;
import com.badlogic.gdx.utils.Array;
import com.badlogic.gdx.utils.Disposable;
import com.badlogic.gdx.utils.Pool;
import com.colorflow.utility.Position;

/**
 * Created by daniele on 09/05/17.
 */

public class ExplosionPool extends Pool<Explosion> {

    private static ExplosionPool instance = null;

    private ExplosionPool() {
    }

    @Override
    protected Explosion newObject() {
        return new Explosion();
    }

    public static ExplosionPool getInstance() {
        if (instance == null) {
            instance = new ExplosionPool();
        }
        return instance;
    }

    public void start(Stage stage, Color color, Position position) {
        Explosion e = obtain();
        stage.addActor(e);
        e.start(color, position);
    }

}

File: core/src/com/colorflow/utility/effect/ExplosionPool.kt
package com.colorflow.utility.effect

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.scenes.scene2d.Stage
import com.badlogic.gdx.utils.Pool
import com.colorflow.utility.Position

/**
 * Created by daniele on 09/05/17.
 */

class ExplosionPool private constructor() : Pool<Explosion>() {

    override fun newObject(): Explosion {
        return Explosion()
    }

    fun start(stage: Stage, color: Color, position: Position) {
        val e = obtain()
        stage.addActor(e)
        e.start(color, position)
    }

    companion object {

        private var instance: ExplosionPool? = null

        fun getInstance(): ExplosionPool {
            if (instance == null) {
                instance = ExplosionPool()
            }
            return instance!!
        }
    }

}

